### ISSUE TITLE
Icon display issues in new version of Glyphicons

### ISSUE THREAD

**Issue by kartik-v** [twbs/bootstrap#16049]
The new glyphicons seem to have newer icons - so its good. However some of the icons in the new set do not have a good resolution at the default sizes. Did not know where to post - since these icons are bundled with bootstrap.

Check an example below for `glyphicon-collapse-down` and `glyphicon-expand`:

Previous Icons rendered this way at a better resolution
![previousglyphicons](https://cloud.githubusercontent.com/assets/3592619/6614986/cca8972e-c8c5-11e4-9989-76fcf253bd24.png)

Current Icons seem to have some issue with the resolution
![currentglyphicons](https://cloud.githubusercontent.com/assets/3592619/6614989/d4acf2c6-c8c5-11e4-94a9-ef590ef821bf.png)

**Comment by cvrebert** [comment]
Most likely an approximate duplicate of #13657.
i.e. It's likely a matter of antialiasing and your choice of font size; we can't really address it at the Bootstrap CSS level, and the Glyphicons guy (jankovarik) doesn't think there's a good way to address it at the font level either.
Also, you neglected to state what OS and browser you're using.

**Comment by kartik-v** [comment]
The above markup is using default bootstrap CSS for rendering the glyphicons without any customization (so the default sizes apply in both cases).

The scenario is consistent across all modern browsers IE11, Firefox, Chrome latest versions and tested on windows OS.

The previous icon set seems to have a consistent resolution display for the default bootstrap glyphicon CSS (font size).

**Comment by cvrebert** [comment]
Just to confirm, you are using Bootstrap v3.3.2, correct?

**Comment by kartik-v** [comment]
Yes. Check these two fiddles I created...

[Bootstrap v3.3.0](http://jsfiddle.net/qtj2botu/5/)
[Bootstrap v3.3.2](http://jsfiddle.net/qtj2botu/)

**Comment by twbs-lmvtfy** [comment]
Hi @kartik-v!

You appear to have posted a live example (http://fiddle.jshell.net/qtj2botu/5/show/light/), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* line 15, column 7: [W013](https://github.com/twbs/bootlint/wiki/W013): Bootstrap version might be outdated. Latest version is at least 3.3.2 ; saw what appears to be usage of Bootstrap 3.3.0

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by cvrebert** [comment]
I am able to confirm the rendering difference in Chrome and IE11 on Windows 8.1.
On the other hand, note that e.g. `glyphicon-folder-close` IMO looks better in 3.3.2 than in 3.3.0.
It is annoying that Bootstrap's default font size is 14px, whereas the Glyphicons are designed for font sizes that are multiples of 12px, but (a) I personally can't think of any backward-compatible way to address this in Bootstrap's CSS (b) it seems unlikely that jankovarik, who has very graciously allowed the gratis inclusion of Glyphicons Halflings in Bootstrap, would be willing to rework them for a 14px-tall pixel grid for free.

On the gripping hand, it is weird for the Glyphicons' display to change like this between non-major releases.
@jankovarik Were there any changes in Glyphicons v1.9 that would explain these display differences?

**Comment by kartik-v** [comment]
I personally feel Glyphicons have the best resolution (when compared to other similar icons from other icon fonts) for bootstrap CSS styling and have always preferred it even when I want to scale it across font sizes. Its the first time I am finding this notableresolution difference in this newer release .

**Comment by jankovarik** [comment]
@kartik-v can you please confirm that this is an issue of the icon-font itself? In other words - in your screenshots in a first post, are you using the same CSS, same browser etc? Is this caused only when you change font files (from version 1.8 to 1.9)? Almost all Halflings were re-created and slightly rounded or shifted by a 1px, in version 1.9, but I kept a same grid:
![screen shot 2015-03-12 at 14 29 21](https://cloud.githubusercontent.com/assets/1397428/6618586/39d730aa-c8c4-11e4-859d-310c51eb2e6f.png)

@cvrebert you're right about default size of icons - Halflings are really made for default size 12px or its multiples. 14px grid as an alternative grid for Bootstrap is something I was thinking about, but I'm not sure about having these two version of font and icons out there, this could be confusing for many Bootstrap users and my customers as well. I'd have to give up 12px version completely and keep only 14px version, but I like them in its current basic size. We'll see.

**Comment by cvrebert** [comment]
@jankovarik 
> can you please confirm that this is an issue of the icon-font itself?

I am reasonably sure there were no relevant CSS changes in Bootstrap v3.3.2. We added classes for the new Glyphicons in bcb90ef3a0affa6540d141733a073593011af064, removed classes for 2 new icons with problematic codepoints in ad138ed885626280be7e9c55b1719b1282e73817, and added WOFF2 in c8855a7ad120b37fd9a25125f90093478656362f, but that's all. We didn't change the CSS for the existing Glyphicons, nor did we change the default `<body>` font-size, and I've reproduced the difference in IE11, [which doesn't support WOFF2](http://caniuse.com/#feat=woff2), so it would indeed seem to be related to the font itself in the existing file formats.

Also, yeah, we totally acknowledge that the sizing conflict is an integration bug, and isn't a bug in Glyphicons itself.

**Comment by jankovarik** [comment]
@cvrebert I see, ok, I'll try to re-generate font files for Halflings during tomorrow - I'll post it here, so you may download it and give it a try ;)

**Comment by kartik-v** [comment]
@jankovarik - answers below
> can you please confirm that this is an issue of the icon-font itself? 
Seems like it - since i am using the default bootstrap CSS and default glyphicons without any customization from my end.

>In other words - in your screenshots in a first post, are you using the same CSS, same browser etc? Is this caused only when you change font files (from version 1.8 to 1.9)?
Yes to both.

**Comment by kartik-v** [comment]
> On the other hand, note that e.g. glyphicon-folder-close IMO looks better in 3.3.2 than in 3.3.0.

I agree - which means we have to retain the good of glyphicons in 3.3.2 and merge it with the good stuff of 3.3.0

**Comment by jankovarik** [comment]
Hey guys, I made two versions - marked them as "a" and "b", each one with different hint option. Here is archive with both versions: http://glyphicons.com/test/archive.zip Let me know if is any better - if yes, I'll use it during next update.

**Comment by kartik-v** [comment]
Thanks @jankovarik.

I tried testing with the versions. Specifically for the two CSS classes `glyphicon-expand` and `glyphicon-collapse-down` did not see any significant change. The v1.8 seems to still render better. Tested it across IE11, Chrome, and Firefox and all have same consistent behavior on windows.
Posting screenshots.

#### Version 1.8 (previous glyphicon release)
![glyphicons_v1 8](https://cloud.githubusercontent.com/assets/3592619/6637677/5d261f14-c9a3-11e4-8ca2-15f6dbda83da.png)

#### Version 1.9a
![glyphicons_v1 9a](https://cloud.githubusercontent.com/assets/3592619/6637680/67147a2a-c9a3-11e4-8c12-0428ac571af6.png)

#### Version 1.9b
![glyphicons_v1 9b](https://cloud.githubusercontent.com/assets/3592619/6637682/6d036338-c9a3-11e4-850a-1a45eaa3c5a9.png)

**Comment by jankovarik** [comment]
It seems like only top part of this specific icon is kind of problematic - I'll try to do some change in vectors, please try to re-download archive with version C and C without hinting (but my guess is that Windows will try to hint it somehow anyway).

If this will not solve it, I'll get back to this issue during monday.

**Comment by cvrebert** [comment]
@kartik-v Have you been able to test version C yet?

**Comment by kartik-v** [comment]
@cvrebert I did not see a version C from @jankovarik to test. Is it there somewhere which I missed?

**Comment by cvrebert** [comment]
@kartik-v Re-download http://glyphicons.com/test/archive.zip It's been updated with version C. See https://github.com/twbs/bootstrap/issues/16049#issuecomment-78940713

**Comment by kartik-v** [comment]
Thanks. I just tested it across browsers on windows. and pasting the result below - unfortunately no major improvement yet when compared to v1.8.
![kartiksglyphiconscompared](https://cloud.githubusercontent.com/assets/3592619/6738609/30d954a8-ce99-11e4-8045-b26a56333d1f.jpg)

**Comment by mdo** [comment]
Closing out for now because v4.

Should there be a Glyphicons update that address this, we'll get the changes in there.
